File: shared/react-native/android/app/src/main/java/io/keybase/ossifrage/modules/StorybookConstants.java
package io.keybase.ossifrage.modules;

import com.facebook.react.bridge.ReactApplicationContext;
import com.facebook.react.bridge.ReactContextBaseJavaModule;

import java.util.HashMap;
import java.util.Map;

import io.keybase.ossifrage.BuildConfig;

public class StorybookConstants extends ReactContextBaseJavaModule {
    public StorybookConstants(final ReactApplicationContext reactContext) {
        super(reactContext);
    }

    @Override
    public Map<String, Object> getConstants() {
        final boolean isStoryBook = BuildConfig.BUILD_TYPE == "storyBook";
        final Map<String, Object> constants = new HashMap<>();
        constants.put("isStorybook", isStoryBook);
        return constants;
    }

    @Override
    public String getName() {
        return "Storybook";
    }
}

File: shared/android/app/src/main/java/io/keybase/ossifrage/modules/StorybookConstants.kt
package io.keybase.ossifrage.modules

import com.facebook.react.bridge.ReactApplicationContext
import com.facebook.react.bridge.ReactContextBaseJavaModule
import io.keybase.ossifrage.BuildConfig

class StorybookConstants(reactContext: ReactApplicationContext?) : ReactContextBaseJavaModule(reactContext) {
    override fun getConstants(): Map<String, Any>? {
        val isStoryBook = BuildConfig.BUILD_TYPE === "storyBook"
        val constants: MutableMap<String, Any> = HashMap()
        constants["isStorybook"] = isStoryBook
        return constants
    }

    override fun getName(): String {
        return "Storybook"
    }
}
